imaginate_api/user/routes.py

```diff
@@ -0,0 +1,98 @@
+from flask import Blueprint, abort, request, redirect, url_for, session, current_app
+from flask_login import current_user, login_user
+from imaginate_api.schemas.user_info import User
+from http import HTTPStatus
+from urllib.parse import urlencode
+import secrets
+import requests
+
+bp = Blueprint("user", __name__)
+
+
+# Initiates the authorization process with the specified provider
+@bp.route("/authorize/<provider>")
+def user_authorize(provider):
+  if not current_user.is_anonymous:
+    return redirect(f'{current_app.config["BASE_URL"]}/?{urlencode(current_user.get_clientside_data())}')
+
+  provider_data = current_app.config["AUTH_PROVIDERS"].get(provider)
+  if not provider_data:
+    abort(
+      HTTPStatus.NOT_FOUND,
+      description=f"Invalid provider, supports: {list(current_app.config["AUTH_PROVIDERS"].keys())}",
+    )
+
+  session["oauth_state"] = secrets.token_urlsafe(32)
+  query = urlencode(
+    {
+      "client_id": provider_data["client_id"],
+      "redirect_uri": url_for("user.user_callback", provider=provider, _external=True),
+      "response_type": "code",  # This tells the OAuth provider that we expect an authorization code to be returned
+      "scope": " ".join(provider_data["scopes"]),
+      "state": session["oauth_state"],
+    }
+  )
+
+  return redirect(f"{provider_data["authorize_url"]}?{query}")
+
+
+# Handles the callback (i.e. redirection response) process with the specified provider
+@bp.route("/callback/<provider>")
+def user_callback(provider):
+  if not current_user.is_anonymous:
+    return redirect(current_app.config["BASE_URL"])
```

Comment:
@faisal-fawad do you think it would make sense to update this redirect like the others?
This would trigger if you accessed the callback endpoint but are already logged in right?
When could that possibly happen?